tasks/main.yml

```diff
@@ -57,20 +57,6 @@
   register: caddy_binary_cache
   tags: skip_ansible_lint
 
-- name: Download Caddy
```

Comment:
The main reason to have this duplicated was for Caddy updates. The role must also be idempotent (only report changes, when something really changes). However I am not fully sure about the logic anymore, because it was quite a long time since I developed this.

The first task is only executed `when` a new release is available. `force` is also `true`.

Reply:
https://github.com/caddy-ansible/caddy-ansible/blob/c9b442887f7c7047d3f225dacae91ee3835fc56b/tasks/main.yml#L31-L39
`caddy_features_cache` is only set when `caddy_update: true`. Meaning that if you set `caddy_update: false` and remove those tasks then Caddy will not be downloaded.